File: app/src/main/java/paul/barthuel/go4lunch/ui/list_view/RestaurantInfoAdapter.java
package paul.barthuel.go4lunch.ui.list_view;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import paul.barthuel.go4lunch.R;
import paul.barthuel.go4lunch.ui.custom_view.CustomRatingBar;

public class RestaurantInfoAdapter extends ListAdapter<RestaurantInfo, RestaurantInfoAdapter.ViewHolder> {

    private Listener listener;

    public RestaurantInfoAdapter(Listener listener) {
        super(new RestaurantInfoDiffCallBack());
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.restaurant_description_item, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int index) {
        viewHolder.bind(getItem(index), listener);
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView mTextViewName;
        private final TextView mTextViewAddress;
        private final TextView mTextViewOpeningHours;
        private final TextView mTextViewDistance;
        private final TextView mTextViewAttendies;
        private final View mRoot;
        private final ImageView mImageViewThumbnail;
        private final CustomRatingBar mCustomRatingBar;



        ViewHolder(@NonNull View itemView) {
            super(itemView);

            mTextViewName = itemView.findViewById(R.id.restaurant_description_item_tv_name);
            mTextViewAddress = itemView.findViewById(R.id.restaurant_description_item_address);
            mTextViewOpeningHours = itemView.findViewById(R.id.restaurant_description_item_schedule);
            mTextViewDistance = itemView.findViewById(R.id.restaurant_description_item_distance);
            mTextViewAttendies = itemView.findViewById(R.id.restaurant_description_item_attendies);
            mRoot = itemView.findViewById(R.id.restaurant_description_item_cl);
            mImageViewThumbnail = itemView.findViewById(R.id.restaurant_description_item_iv_thumbnail);
            mCustomRatingBar = itemView.findViewById(R.id.restaurant_description_item_crb_rating);
        }

        private void bind(final RestaurantInfo restaurantInfo, final Listener listener) {

            mTextViewName.setText(restaurantInfo.getName());
            mTextViewAddress.setText(restaurantInfo.getAddress());
            mTextViewOpeningHours.setText(restaurantInfo.getOpeningHours());
            mTextViewDistance.setText(restaurantInfo.getDistance());
            if (restaurantInfo.isAttendiesVisible()) {
                mTextViewAttendies.setText(restaurantInfo.getAttendies());
                mTextViewAttendies.setVisibility(View.VISIBLE);
            }else {
                mTextViewAttendies.setVisibility(View.INVISIBLE);
            }
            mCustomRatingBar.setStars(restaurantInfo.getRating());
            Log.d("courgette", "bind: name " + restaurantInfo.getName() + "rating " + restaurantInfo.getRating());
            Glide.with(mImageViewThumbnail).load(restaurantInfo.getImage()).into(mImageViewThumbnail);
            mRoot.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.onRestaurantInfoClick(restaurantInfo);
                }
            });
        }
    }

    public interface Listener {
        void onRestaurantInfoClick(RestaurantInfo restaurantInfo);
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/list_view/RestaurantInfoAdapter.kt
package paul.barthuel.go4lunch.ui.list_view

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import paul.barthuel.go4lunch.R
import paul.barthuel.go4lunch.ui.custom_view.CustomRatingBar

class RestaurantInfoAdapter(private val listener: Listener) : ListAdapter<RestaurantInfo, RestaurantInfoAdapter.ViewHolder>(RestaurantInfoDiffCallBack()) {
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.restaurant_description_item, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, index: Int) {
        viewHolder.bind(getItem(index), listener)
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val mTextViewName = itemView.findViewById<TextView>(R.id.restaurant_description_item_tv_name)
        private val mTextViewAddress = itemView.findViewById<TextView>(R.id.restaurant_description_item_address)
        private val mTextViewOpeningHours = itemView.findViewById<TextView>(R.id.restaurant_description_item_schedule)
        private val mTextViewDistance = itemView.findViewById<TextView>(R.id.restaurant_description_item_distance)
        private val mTextViewAttendies = itemView.findViewById<TextView>(R.id.restaurant_description_item_attendies)
        private val mRoot = itemView.findViewById<View>(R.id.restaurant_description_item_cl)
        private val mImageViewThumbnail = itemView.findViewById<ImageView>(R.id.restaurant_description_item_iv_thumbnail)
        private val mCustomRatingBar = itemView.findViewById<CustomRatingBar>(R.id.restaurant_description_item_crb_rating)
        private val mBackground = itemView.findViewById<View>(R.id.restaurant_description_item_background)
        fun bind(restaurantInfo: RestaurantInfo, listener: Listener) {
            mTextViewName.text = restaurantInfo.name
            mTextViewAddress.text = restaurantInfo.address
            mTextViewOpeningHours.text = restaurantInfo.openingHours
            mTextViewDistance.text = restaurantInfo.distance
            if (restaurantInfo.isAttendiesVisible) {
                mTextViewAttendies.text = restaurantInfo.attendies
                mTextViewAttendies.visibility = View.VISIBLE
            } else {
                mTextViewAttendies.visibility = View.INVISIBLE
            }
            mCustomRatingBar.setStars(restaurantInfo.rating)
            Log.d("courgette", "bind: name " + restaurantInfo.name + "rating " + restaurantInfo.rating)
            Glide.with(mImageViewThumbnail).load(restaurantInfo.image).into(mImageViewThumbnail)
            mRoot.setOnClickListener { listener.onRestaurantInfoClick(restaurantInfo) }
            mBackground.setBackgroundResource(restaurantInfo.backgroundColor)
        }

    }

    interface Listener {
        fun onRestaurantInfoClick(restaurantInfo: RestaurantInfo?)
    }

}